x-pack/plugin/profiling/src/test/java/org/elasticsearch/xpack/profiling/HostMetadataTests.java

```diff
@@ -38,4 +37,85 @@ public void testCreateFromRegularSource() {
         assertEquals(region, host.instanceType.region);
         assertEquals(instanceType, host.instanceType.name);
     }
+
+    public void testCreateFromSourceGCP() {
+        final String hostID = "1440256254710195396";
+        final String machine = "x86_64";
+        final String provider = "gcp";
+        final String[] regions = { "", "", "europe-west1", "europewest", "europe-west1" };
+        final String[] zones = {
+            "",
```

Comment:
Can we please also add a case for random garbage (e.g. `randomAlphaOfLength(5)`)?